### PR TITLE
Build clad

### PR DESCRIPTION
**Pull request by mattgodbolt**
CC @vgvassilev

### CODE REVIEW

File: clad/build.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    VERSION=trunk-$(date +%Y%m%d)
+    BRANCH=master
+    REMOTE=heads/${BRANCH}
+else
+    BRANCH=v${VERSION}
+    REMOTE=tags/${BRANCH}
+fi
+
+URL=https://github.com/vgvassilev/clad
+CLANG_VERSION=18.1.0
```
Comment (partouf) [inline clad/build.sh:17]: I would make this configurable and include it in the S3 name so we can support the plugin for multiple clang versions

Comment (Vipul-Cariappa) [reply clad/build.sh:17]: +1

Comment (mattgodbolt) [reply clad/build.sh:17]: I'll take a look at that.

**Comment by mattgodbolt** [comment]
Something like that yes: if you look at the clangs or similar you'll see how they do it.

We'd also need it to be added to our "daily builds" -> https://github.com/compiler-explorer/compiler-workflows - I can help with either of these.

**Comment by Vipul-Cariappa** [comment]
Hi. I have opened PRs at https://github.com/compiler-explorer/compiler-explorer/pull/7311 and https://github.com/compiler-explorer/compiler-workflows/pull/31 to add the nightly build of clad.
I am not sure about the changes required in https://github.com/compiler-explorer/infra.

**Comment by mattgodbolt** [comment]
First run: https://github.com/compiler-explorer/compiler-workflows/actions/runs/12993976934

Changes in infra; just adding a "nightly" equivalent build (see one of the other nightly compilers). I can help if you can't make any sense of the setup (it's not well documented)

**Pull request merged**